### PR TITLE
Added support for the archunit

### PR DESCRIPTION
**Pull request by dariuszzbyrad**
Signed-off-by: Dariusz Zbyrad <dariusz.zbyrad@gmail.com>

## What changes were proposed in this pull request?

Added support for the ArchUnit framework. (https://www.archunit.org/)

## How was this patch tested?
Run existing test and build packages. Added new test.

## Quick checklist

The following checklist helps ensure your PR is complete:

- [X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X] Created tests for any significant new code additions.
- [X] Relevant tests for your changes are passing.

### CODE REVIEW

File: datavec/datavec-api/src/test/java/org/datavec/api/transform/ops/AggregableMultiOpArchTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Created by dariuszzbyrad on 7/31/2020.
+ */
+@RunWith(ArchUnitRunner.class)
+@AnalyzeClasses(packages = "org.datavec.api.transform.ops", importOptions = {ImportOption.DoNotIncludeTests.class})
+public class AggregableMultiOpArchTest extends BaseND4JTest {
+
+    @ArchTest
+    public static final ArchRule ALL_AGGREGATE_OPS_MUST_BE_SERIALIZABLE = classes()
+            .that().resideInAPackage("org.datavec.api.transform.ops")
+            .and().doNotHaveSimpleName("AggregatorImpls")
```
Comment (dariuszzbyrad) [inline datavec/datavec-api/src/test/java/org/datavec/api/transform/ops/AggregableMultiOpArchTest.java:25]: ArchUnit does't contain method like `doNotHaveSimpleName(Collection<String> names)`, but I'll push PR to add it and when it'll be acceptable, I'll refactor it.

File: datavec/datavec-api/src/test/java/org/datavec/api/transform/ops/AggregableMultiOpArchTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Created by dariuszzbyrad on 7/31/2020.
+ */
+@RunWith(ArchUnitRunner.class)
+@AnalyzeClasses(packages = "org.datavec.api.transform.ops", importOptions = {ImportOption.DoNotIncludeTests.class})
+public class AggregableMultiOpArchTest extends BaseND4JTest {
+
+    @ArchTest
+    public static final ArchRule ALL_AGGREGATE_OPS_MUST_BE_SERIALIZABLE = classes()
```
Comment (dariuszzbyrad) [inline datavec/datavec-api/src/test/java/org/datavec/api/transform/ops/AggregableMultiOpArchTest.java:23]: Potential error looks like:
_Class <org.datavec.api.transform.ops.DispatchOpTest> does not implement java.io.Serializable_

**Comment by agibsonccc** [comment]
@dariuszzbyrad can you clarify the intent of this PR a bit? What sort of benefits do you see in adding this? Thanks!

**Comment by agibsonccc** [comment]
@dariuszzbyrad  this looks great. I looked at your test and understand why you want to add it. This would be great for maintainability purposes. Thanks!

**Comment by dariuszzbyrad** [comment]
@agibsonccc Should I add/change something else or it can be merged?

**Review by agibsonccc — APPROVED** [review]

**Pull request merged**